File: 2018-19/unit6/cash/cash.py
'''
    Tahmidul Alam
    CS50AP
    Cash
    cash.py
'''

from cs50 import get_float

QTR = 25
DIM = 10
NIC = 5


def main():
    # Validate user input
    while True:
        print("Change: ", end="")
        user_input = round(get_float() * 100)
        if user_input >= 0:
            break
    # Declare a variable for counting coins
    coins = 0

    # Get the whole number of quarters from the change
    quarters = user_input // QTR

    # Find the remaining value after quarters
    user_input %= QTR

    # Get the whole number of dimes from the remaining change
    dimes = user_input // DIM

    # Find the remaining value after dimes
    user_input %= DIM

    # Get the whole number of nickels from the remaining change
    nickels = user_input // NIC

    # The rest of the value must be pennies
    user_input %= NIC

    # Add up the amount of coins
    coins = user_input + quarters + nickels + dimes

    print(coins)


if __name__ == "__main__":
    main()

File: 2018-19/pset1/cash/cash.c
/**
 * cash.c
 *
 * PSET1
 * CS50AP
 *
 * Tahmidul Alam
 *
*/
#include <cs50.h>
#include <stdio.h>
#include <math.h>

// Coin constants
#define QTR 25
#define DIM 10
#define NIC 5

int main(void)
{
    // Get User Input until it's not negative
    int user_input;
    do
    {
        user_input = roundf(get_float("Change owed: ") * 100.0);
    }
    while (user_input < 0);

    int coins = 0;

    // Finding the whole number of quarters
    int quarters = user_input / QTR;

    // Remaining Coins (Not quarters)
    user_input %= QTR;

    // Whole number of dimes
    int dimes = user_input / DIM;

    // Remaining Coins (Not Dimes)
    user_input %= DIM;

    // Whole number of nickels
    int nickels = user_input / NIC;

    // Rest of the coins
    user_input %= NIC;

    // Adding the number of coins
    coins = user_input + quarters + nickels + dimes;

    // What the user sees
    printf("%i\n", coins);
}